File: example/foo-wrapper.rs
use foo_command::{export, exports::wasi::cli::run::Guest};
use foo_command::test::foo::custom::words;

struct RunnerImpl;

impl Guest for RunnerImpl {
    fn run() -> Result<(), ()> {
        let the_words = words("Are these words?");

        assert!(the_words.len() == 3);

        Ok(())
    }
}

export!(RunnerImpl);

File: example/go/foo-wrapper.go
// TinyGo currently can only compile Wasm components that implement `wasi:cli/run`,
// which means it needs to have package "main" and implement a `main` function.
// To implement a reactor component, just use an empty `main` function.
// https://github.com/tinygo-org/tinygo/issues/4843
package main

import (
	"example.com/wit/test/foo/custom"

	"this-is.a/dependency"
)

func main() {
	dependency.DoSomething(func(s string) []string {
		return custom.Words(s).Slice()
	})
}
